Base РК 2

tUnknownLegend opened the pull request.

EuphoriaAbsorber committed:
changed url for getproducts

EuphoriaAbsorber committed:
changed path for get products

tUnknownLegend committed:
Шрифты вынесены в индекс

так же шрифты теперь какие надо, но это не точно :/

tUnknownLegend committed:
перенос шрифтов

thelvv committed:
сделана страница пользователя

thelvv committed:
Merge branch 'main' into Base-2.2_profile_page

"updated with main"

thelvv committed:
почти закончена логика, осталось допилить запросы и реакцию на действия пользователя

thelvv committed:
основы базовых классов

thelvv committed:
изменения для прохождения линтеров

tUnknownLegend committed:
добавлен базовый store

thelvv committed:
исправлен gitignore

thelvv committed:
исправлено название файла и gitignore

Liza1040 committed:
Добавлен попап для изменения данных пользователя

Liza1040 committed:
Поправлены ошибки, найденные линтером

tUnknownLegend committed:
почти сверстана карта товара в корзине

Liza1040 committed:
Данные со страницы профиля передаются в попап

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Добавлены слушатели для удаления банковской карты и адреса

tUnknownLegend committed:
сверстана корзина

Liza1040 committed:
Добавлен попап для добавления банковской карты

Liza1040 committed:
Поправлены ошибки, найденные линтером

tUnknownLegend committed:
CSP

tUnknownLegend committed:
обновлени CSP

tUnknownLegend committed:
+ base store & view

tUnknownLegend committed:
Merge branch 'main' into Base-2.12_cart_page

tUnknownLegend committed:
main merge

thelvv committed:
верстка и базовая логика страницы каталога

Liza1040 committed:
Добавлен попап для добавления/редактирования адреса

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Поправлены ошибки, описанные в комментариях к ПР

Liza1040 committed:
Поправлены ошибки, найденные линтером

tUnknownLegend committed:
Merge branch 'main' into Base-2.12_catalog_page

tUnknownLegend committed:
merge fix

tUnknownLegend committed:
Merge branch 'main' into Base-2.9_Base_View+Store

Liza1040 committed:
Добавлены попапы для редактирования адреса и банковской карты

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Фикс линтера

Liza1040 committed:
Фикс линтера

tUnknownLegend committed:
исправления верстки и css после мерджа

thelvv committed:
добавлено изменение кнопок в корзину по клику, включены топ-категории на главной

thelvv committed:
добавлено изменение кнопок в корзину по клику, включены топ-категории на главной, исправлены ошибки на тестах

tUnknownLegend committed:
+ get user

tUnknownLegend committed:
исправление get user

tUnknownLegend committed:
работают попапы, нужна логика

tUnknownLegend committed:
добавил label в popUp

Liza1040 committed:
Добавлен акцент на выбранные адреса и карты

tUnknownLegend committed:
user store для авторизации

tUnknownLegend committed:
испрваление рефреша хедера

tUnknownLegend committed:
испрваление рефреша хедера 2

tUnknownLegend committed:
испрваление рефреша хедера final

Liza1040 committed:
Добавлен bind в addEventListener

thelvv committed:
на главную и на страницу категории добавлены главные listener'ы, добавлена обработка кликов

Liza1040 committed:
Добавлено получение данных заказа со страницы

tUnknownLegend committed:
изменение данных пользователя

tUnknownLegend committed:
доделываю страницу пользвателя...

tUnknownLegend committed:
рабочий профиль

осталось совместить

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Поправлены ошибки, найденные линтером

tUnknownLegend committed:
переписал main page  через store

tUnknownLegend committed:
исправление null полей

tUnknownLegend committed:
исправление null полей

tUnknownLegend committed:
исправление null полей 2

tUnknownLegend committed:
исправление null полей 3

tUnknownLegend committed:
исправление null полей 4

tUnknownLegend committed:
исправление null полей 5

tUnknownLegend committed:
исправление null полей 6

tUnknownLegend committed:
исправление null полей 7

tUnknownLegend committed:
исправление null полей 7

tUnknownLegend committed:
исправление null полей 8

tUnknownLegend committed:
исправление null полей 9

tUnknownLegend committed:
исправление null полей 10

tUnknownLegend committed:
надеюсь главная работат -_-

tUnknownLegend committed:
работает?

tUnknownLegend committed:
работает^/

tUnknownLegend committed:
работает^|^

tUnknownLegend committed:
работает _|_

tUnknownLegend committed:
работает _|_

thelvv committed:
добавлена основа бесконечного скролла

tUnknownLegend committed:
1

tUnknownLegend committed:
Merge branch 'Base-2.12_catalog_page' into Base-2.13-merge

tUnknownLegend committed:
fake

Liza1040 committed:
Добавлено изменение адреса в корзине

Liza1040 committed:
Добавлена обработка увеличения/уменьшения количества товара в корзине, передача id выбранного адреса/карты в попап и обратно

Liza1040 committed:
Добавлен подсчет итоговых стоимостей при увеличении/уменьшении количества товара. Добавлено изменение карты в блоке оформления заказа

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Поправлены ошибки, найденные линтером

Liza1040 committed:
Фикс линтера

Liza1040 committed:
Поправлен сбор данных для оформления заказа

tUnknownLegend committed:
поправил кнопки

EuphoriaAbsorber committed:
load avatars

EuphoriaAbsorber committed:
deploy settings for queries

EuphoriaAbsorber committed:
css fixes for avatar change popup

tUnknownLegend committed:
начал роутер

tUnknownLegend committed:
merge

Liza1040 committed:
Merge remote-tracking branch 'origin/Base-2.25_merging_usprofile_with_back' into Base-2.12_cart_page

Liza1040 committed:
Fix merge

Liza1040 committed:
Поправлены ошибки, найденные линтером

tUnknownLegend committed:
сделана загрузка товаров по категории

tUnknownLegend committed:
поправил передачу контекста при авторизации

tUnknownLegend committed:
поправлен  переход по истории

так же сделаны динамические названия вкладок

Liza1040 committed:
Добавлены функции получения корзины, удаления элемента и очищения корзины

Liza1040 committed:
Поправлена функция очищения корзины в store. Функции выбора даты и времени доставки объединены в одну

Liza1040 committed:
Поправлен сбор данных корзины для отправки на сервер

tUnknownLegend committed:
исправление отображениея аватара

tUnknownLegend committed:
Merge branch 'Base-2.12_cart_page' into Base-2.14_merge_cart

tUnknownLegend committed:
исправил стили

thelvv committed:
полностью исправлена валидация данных пользователя на userpage

Liza1040 committed:
Поправлены обработчики на странице корзины. Добавлена поддержка галочек

thelvv committed:
поправлены топ-категории, сделана выпадашка на кнопке Каталог

Liza1040 committed:
Добавлен вызов функций для отправки на бек данных при изменении количества товаров и сами функции. Поправлен запрос удаления товара.

Liza1040 committed:
Добавлена обработка создания заказа и отправки запроса на сервер.

Liza1040 committed:
Данные корзины берутся из стора

Liza1040 committed:
Дааные получаются из сторов

Liza1040 committed:
В попапе выбора карты и адреса добавлена кнопка "Добавить"

tUnknownLegend committed:
изменения на странице пользвоателя

tUnknownLegend committed:
Merge branch 'Base-2.14_bugfixes' into Base-2.14_merge_cart

tUnknownLegend committed:
исрпавление dropdown'а и стилей

tUnknownLegend committed:
поправил верстку на главной странице

tUnknownLegend committed:
Merge branch 'Base-2.12_cart_page' into Base-2.14_merge_cart

Liza1040 committed:
Добавлена оплата наличными при получении

tUnknownLegend committed:
соединение каталога и корзины

tUnknownLegend committed:
добавление в корзину работатет

tUnknownLegend committed:
исрпавления запросов на странице пользователя

tUnknownLegend committed:
оформление заказа

tUnknownLegend committed:
убал моки

tUnknownLegend committed:
исрпавления полей пользователя

tUnknownLegend committed:
соединяю пользователя 1

tUnknownLegend committed:
соединяю пользователя 2

tUnknownLegend committed:
соединяю пользователя 3

tUnknownLegend committed:
соединяю пользователя 4

tUnknownLegend committed:
соединяю пользователя 5

tUnknownLegend committed:
соединяю пользователя 6

tUnknownLegend committed:
соединяю пользователя 7

tUnknownLegend committed:
соединяю пользователя 8

tUnknownLegend committed:
соединяю пользователя 9

tUnknownLegend committed:
соединяю пользователя 10

tUnknownLegend committed:
соединяю пользователя 11

tUnknownLegend committed:
соединяю пользователя 1

tUnknownLegend committed:
соединяю пользователя 11

tUnknownLegend committed:
соединяю пользователя 12

tUnknownLegend committed:
соединяю пользователя 13

tUnknownLegend committed:
соединяю корзину 1

tUnknownLegend committed:
соединяю корзину 2

tUnknownLegend committed:
соединяю корзину 3

tUnknownLegend committed:
исрпавление регистрации

tUnknownLegend committed:
исправление загрузки каталога

tUnknownLegend committed:
сделал таймер

tUnknownLegend committed:
валидация при добовлении карт карт

tUnknownLegend committed:
валидация добовления карт

tUnknownLegend committed:
Merge branch 'Base-2.12_cart_page' into Base-2.14_merge_cart

tUnknownLegend committed:
поправил валидацию карт

tUnknownLegend committed:
поправил добовление карт

tUnknownLegend committed:
исправил adress

tUnknownLegend committed:
совмещая профиль 1

tUnknownLegend committed:
совмещая профиль 2

tUnknownLegend committed:
поправил id

tUnknownLegend committed:
поправил id

tUnknownLegend committed:
поправил id 2

tUnknownLegend committed:
исравил загрузку карт

tUnknownLegend committed:
корзина + проифль 1

tUnknownLegend committed:
изменил парсинг id на юзере

tUnknownLegend committed:
починка карт профиля

tUnknownLegend committed:
надеюсь корзина работает

tUnknownLegend committed:
уброл мок

tUnknownLegend committed:
вернул prevent Default в корзине

tUnknownLegend committed:
оформлние заказа 1

tUnknownLegend committed:
оформлние заказа 2

tUnknownLegend committed:
поправил изменение адреса

tUnknownLegend committed:
linters

tUnknownLegend committed:
убрал мок

tUnknownLegend committed:
вернул csp

tUnknownLegend committed:
исрпавил csp

tUnknownLegend committed:
исправиление корзины

tUnknownLegend committed:
исправление выбора оплаты

tUnknownLegend committed:
поправил верстку

tUnknownLegend committed:
исправление корзины

tUnknownLegend committed:
исправление передачи id при оформлении заказа

tUnknownLegend committed:
исправление оформления заказа

tUnknownLegend committed:
исправление оформления заказа

tUnknownLegend committed:
совмещение локальной корзины с облочной

tUnknownLegend committed:
совмещение локальной корзины с облочной 2

tUnknownLegend committed:
поправил удаление из корзины

tUnknownLegend committed:
соединяем 1

tUnknownLegend committed:
поправил merge корзины

tUnknownLegend committed:
совмещение новой карты оплаты

tUnknownLegend committed:
исправление парсинга карты 2

tUnknownLegend committed:
исрпавление карт

tUnknownLegend committed:
сделал более общие функции в юзер сторе

tUnknownLegend committed:
поправил путь запроса логина

tUnknownLegend committed:
рефакторинг

tUnknownLegend committed:
убрал мок

tUnknownLegend committed:
чиню загрузку карты

tUnknownLegend committed:
чиню загрузку карты 2

tUnknownLegend committed:
поправил верстку каталога каталог

tUnknownLegend committed:
сортировки

tUnknownLegend committed:
исрпавление добавления карты

tUnknownLegend committed:
исправление добавления карты

tUnknownLegend committed:
исправление добавления карты 3

tUnknownLegend committed:
исправление добавления карты 4

tUnknownLegend committed:
исправление добавления карты 5

tUnknownLegend committed:
png -> webp

DPeshkoff commented on .gitignore:
```diff
@@ -1,4 +1,4 @@
 node_modules
-public/src/*/templates.js
 dist
+#public/index.hbs
```
кто такой?

DPeshkoff commented on public/index.hbs:
```diff
@@ -3,8 +3,8 @@
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <meta http-equiv="Content-Security-Policy"
-          content="default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self' img.mvideo.ru; style-src 'self' 127.0.0.1:8081/;base-uri 'self'; form-action 'self'; font-src 'self' fonts.gstatic.com; manifest-src 'self'; upgrade-insecure-requests">
+<!--    <meta http-equiv="Content-Security-Policy"-->
+<!--          content="default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self' img.mvideo.ru; style-src 'self' 'unsafe-inline';base-uri 'self'; form-action 'self'; font-src 'self' fonts.gstatic.com; manifest-src 'self'; upgrade-insecure-requests">-->
```
Давайте починим, а не будем выпиливать

DPeshkoff replied:
а чего оно вернулось?

DPeshkoff commented on public/index.scss:
```diff
@@ -1,215 +1,261 @@
+  margin: 0;
+}
+
+input[type=number] {
```
что за стили на элементы?

DPeshkoff commented on public/src/actions/cart.js:
```diff
@@ -0,0 +1,114 @@
+     */
+    makeOrder(data) {
+        Dispatcher.dispatch({
+            actionName: CartActionTypes.MAKEORDER,
```
```suggestion
            actionName: CartActionTypes.MAKE_ORDER,
```

DPeshkoff commented on public/src/actions/cart.js:
```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import Dispatcher from '../modules/dispatcher.js';
```
.js?

DPeshkoff commented on public/src/actions/itemCards.js:
```diff
@@ -1,49 +1,81 @@
 'use strict';
 
-import Dispatcher from '../modules/dispatcher.js';
+import Dispatcher from '../modules/dispatcher';
```
а тут без .js...

DPeshkoff commented on public/src/actions/profile.js:
```diff
@@ -34,74 +32,43 @@ export const ProfileAction = {
+     * @param {object} data - отредактированное поле
      */
-    saveEditData(field, newData) {
+    saveEditData( data) {
```
что за пробел

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+     * @param {object} event - событие
+     */
+    async listenClickEditeAddress(event) {
+        const cardID = event.target.id.replace('edit-img-', '');
```
а не проще в дата-атрибут положить? 

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+     * адреса
+     * @param {object} event - событие
+     */
+    async listenClickEditeAddress(event) {
```
почему async?

DPeshkoff replied:
```suggestion
    async listenClickEditAddress(event) {
```

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+     * адреса
+     * @param {object} event - событие
+     */
+    async listenClickAddAddress(event) {
```
почему async?

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+                key.addEventListener('click', this.listenClickDeleteAddress);
+            });
+        } else {
+            console.log('element not found', deleteAddress);
```
можно без лога

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+                key.removeEventListener('click', this.listenClickEditeAddress);
+            });
+        } else {
+            console.log('element not found', editAddress);
```
можно без лога

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+        if (newAddress) {
+            newAddress.addEventListener('click', this.listenClickAddAddress);
+        } else {
+            console.log('element not found', newAddress);
```
можно без лога

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+            if (newAddress) {
+                newAddress.addEventListener('click', this.listenClickAddAddress);
+            } else {
+                console.log('element not found', newAddress);
```
можно без лога

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+                key.addEventListener('click', this.listenClickEditeAddress);
+            });
+        } else {
+            console.log('element not found', editAddress);
```
можно без лога

DPeshkoff commented on public/src/components/AddressCard/AddressCard.scss:
```diff
@@ -0,0 +1,89 @@
+
+//.address-card:hover {
+//    filter: blur(1px);
+//}
```
кто такой?

DPeshkoff commented on public/src/components/AddressCard/AddressCard.js:
```diff
@@ -0,0 +1,144 @@
+    /**
+     * Метод, отрисовывающий компонент в родительский HTML-элемент по заданному шаблону,
+     * импортированному из templates.js
+     * @param {context} context, с учетом которого будет произведен рендер
```
Не знаю такого типа данных, как context

DPeshkoff commented on public/src/components/AddressCard/addressCard.hbs:
```diff
@@ -0,0 +1,42 @@
+                    {{this.street}}
+                </div>
+
+                <div class="address-card-city text-normal-default"> д. {{this.house}}, {{#if this.falt}}кв. {{this.flat}}{{/if}}</div>
```
falt или flat?

DPeshkoff commented on public/src/components/CartItem/CartItem.hbs:
```diff
@@ -0,0 +1,55 @@
+                        В избранном {{else}} В избранное {{/if}}</span>
+                </label>
+                <button class="button-default-no-box delete-cart-item__down item_cart__selection link__no-color" id="button-default_cart__delete">
+                    <img src="/img/Bin.svg" alt="delete picture" class="button-delete_cart__img">
```
неправильный alt

DPeshkoff commented on public/src/components/CartItem/CartItem.hbs:
```diff
@@ -0,0 +1,55 @@
+
+        <div class="count-selector">
+            <button class="button-notext">
+                <span class="link__no-color" id="button-minus_cart/{{this.id}}">−</span>
```
страшные айдишники
почему не стали делать дата-атрибуты?

tUnknownLegend replied:
Мы попробовли сделать через id, чтобы не использовать querySelector, решение оказалось спорным, но рабочим

tUnknownLegend replied:
Надо бы переделать, но когда -_-

DPeshkoff commented on public/src/components/CartItem/CartItem.js:
```diff
@@ -0,0 +1,34 @@
+     */
+    prepareCategory(context) {
+        return {item: {...context}};
+    }
```
в базовом компоненте тот же самый код, в чем смысл?

DPeshkoff commented on public/src/components/CartItem/CartItem.scss:
```diff
@@ -0,0 +1,84 @@
+}
+
+.delete-cart-item__down {
+}
```
зачем?

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.hbs:
```diff
@@ -0,0 +1,101 @@
+            <a href="/item/{{this.id}}">
+                <img src="{{this.imgsrc}}" class="catalog-item-card__pic-block__pic" alt="item_photo"
+                     id="catalog-item-pic"
+                     data-href="/item/{{id}}">
```
this.id или id?

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.hbs:
```diff
@@ -0,0 +1,101 @@
+    <div class="catalog-item-card">
+        <div class="catalog-item-card__pic-block" id="catalog_item-pic-block">
+            <a href="/item/{{this.id}}">
+                <img src="{{this.imgsrc}}" class="catalog-item-card__pic-block__pic" alt="item_photo"
```
alt некорректный

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.hbs:
```diff
@@ -0,0 +1,101 @@
+                <div class="catalog-item-card-properties-block__rating-block">
+                    <img src="/img/star-icon.svg"
+                         class="catalog-item-card__description-block__properties-block__rating-block__star-icon"
+                         alt="star_icon">
```
alt

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.hbs:
```diff
@@ -0,0 +1,101 @@
+                <div class="text-normal-default catalog-item-card__properties-block__property-line">
+                    {{#if this.propertyName4}}
+                        {{this.propertyName4}}: {{this.property4}}
+                    {{/if}}
```
что за прибитые молотком propertyName1,2,3,4? не хотите сделать полноценную шаблонизацию?

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.hbs:
```diff
@@ -0,0 +1,101 @@
+
+                <label class="item_cart__selection disabled">
+                    <input type="checkbox" class="favourite-opt " id="catalog_like-button/{{this.id}}">
+                    <!--                        {{#if this.favourite}} checked {{/if}}>-->
```
не понял, куда выпилили

DPeshkoff commented on public/src/components/CatalogItemCard/CatalogItemCard.js:
```diff
@@ -0,0 +1,50 @@
+     */
+    prepareRenderData(context) {
+        return {catalogCardItem: {...context}};
+    }
```
странное переопределение базового класса, если честно
вам принципиально нужен catalogCardItem?

DPeshkoff commented on public/src/components/Footer/footer.hbs:
```diff
@@ -1,40 +1,60 @@
+                <a href="https://github.com/EuphoriaAbsorber"><img src="/img/Github_logo.webp"
+                                                                   class="footer__contact-logo footer__margin-contact-logo"
+                                                                   alt="GitHubLogo"></a>
             </div>
```
можно разбить на подкомпоненты

DPeshkoff commented on public/src/components/Footer/footer.hbs:
```diff
@@ -1,40 +1,60 @@
+                                                     alt="TGLogo"></a>
+                <a href="https://github.com/Liza1040"><img src="/img/Github_logo.webp"
+                                                           class="footer__contact-logo footer__margin-contact-logo"
+                                                           alt="GitHubLogo"></a>
```
альт тоже не отражает сути)

DPeshkoff commented on public/src/components/Form/form.scss:
```diff
@@ -42,3 +42,8 @@
+
+.login__form__error {
+    margin-left: 5px;
+    margin-top: 10px;
```
опять что-то с кратностью отступов
в файлах выше, кстати, тоже, прошерстите весь код плиз

DPeshkoff commented on public/src/components/Header/Header.js:
```diff
@@ -6,6 +6,39 @@ import './header.scss';
  */
 export default class Header extends BaseComponent {
+    context = {
+        categories: {
+            Smartphone: {
+                nameCategory: 'Телефоны',
+                href: '/category/phones',
+            },
+            Computer: {
+                nameCategory: 'Компьютеры',
+                href: '/category/computers',
+            },
+            Monitors: {
+                nameCategory: 'Мониторы',
+                href: '/category/monitors',
+            },
+            TV: {
+                nameCategory: 'Телевизоры',
+                href: '/category/televisors',
+            },
+            Watch: {
+                nameCategory: 'Часы',
+                href: '/category/watches',
+            },
+            Tablet: {
+                nameCategory: 'Планшеты',
+                href: '/category/tablets',
+            },
+            Accessories: {
+                nameCategory: 'Аксессуары',
+                href: '/category/accessories',
+            },
+        },
```
а почему не с бэка? 

tUnknownLegend replied:
Категории не меняются - нет смысла зря делать запрос

tUnknownLegend replied:
либо здесь либо в сторе можно удалить, мб лучше в сторе, потому что только здесь это нужно

DPeshkoff commented on public/src/components/Footer/footer.scss:
```diff
@@ -36,3 +37,8 @@
     margin-left: 20px;
 }
+
+.logo-img {
```
у вас разве не logo-reazon?

thelvv replied:
Это для лого ВК, TG и гитхаба около имен членов команды

DPeshkoff commented on public/src/components/Header/header.scss:
```diff
@@ -1,157 +1,239 @@
-    width: 100px;
-    height: 40px;
-    align-self: center;
+  margin-left: 20px;
```
го stylelint

tUnknownLegend replied:
настроил, только в другой ветке уже, мб сюда и не вмержим уже, но скорее вмержим

DPeshkoff commented on public/src/components/Header/header.scss:
```diff
@@ -1,157 +1,239 @@
+  border-radius: 10px;
+}
+
+/* Links inside the dropdown */
+.dropdown-content a {
```
поймал на элементы
откуда брали?)

thelvv replied:
Из примеров dropdown'ов

tUnknownLegend replied:
мы попозже уберем завязку не элементы, в одном из следующих МРов он по-другому сделан в целом

DPeshkoff commented on public/src/components/Header/header.scss:
```diff
@@ -1,157 +1,239 @@
+  top: 80px;
+  padding: 16px;
+  animation: animation_fade-out 5s;
+  z-index: 999999;
```
жесть

DPeshkoff commented on public/src/components/ItemCard/itemCard.hbs:
```diff
@@ -1,34 +1,56 @@
+             data-count="{{this.count}}"
+            {{#if this.count}} style="display: grid;" {{/if}}>
+            <button class="button-notext">
+                <span class="link__no-color text-normal-large" ind="itemcard_button-minus_cart/{{this.id}}">−</span>
```
ind?

DPeshkoff replied:
так что за ind?

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+     * @param {object} event - событие
+     */
+    async listenClickDeletePaymentCard(event) {
+        profileAction.deleteCard(Number(event.target.id.replace('delete-img-paymentCard/', '')));
```
опять что-то страшное
см выше

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+                key.addEventListener('click', this.listenClickDeletePaymentCard);
+            });
+        } else {
+            console.log('element not found', paymentCard);
```
лишнее

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+            if (newPaymentCard) {
+                newPaymentCard.addEventListener('click', this.listenClickAddPaymentCard);
+            } else {
+                console.log('element not found', newPaymentCard);
```
лишнее

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+        if (newPaymentCard) {
+            newPaymentCard.removeEventListener('click', this.listenClickAddPaymentCard);
+        } else {
+            console.log('element not found', newPaymentCard);
```
лишнее

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.scss:
```diff
@@ -0,0 +1,88 @@
+    width: 40px;
+    align-items: center;
+    //grid-template-columns: 1fr;
+    //display:grid;
```
Почему закомментировано? 

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.scss:
```diff
@@ -0,0 +1,88 @@
+    color: var(--text-color);
+}
+
+.card-title-main {
```
названия классов в этом файле не соответствуют БЭМу, перепишите, пожалуйста

DPeshkoff commented on public/src/components/PaymentCard/PaymentCard.scss:
```diff
@@ -0,0 +1,88 @@
+    height: 32px;
+    align-self: center;
+    justify-self: center;
+    margin-top: -4px;
```
похоже на костыль

DPeshkoff commented on public/src/components/PaymentCard/paymentCard.hbs:
```diff
@@ -0,0 +1,54 @@
+
+            <div class="change-payment-card change-pop-up_user-page" id="change-payment-card-{{this.id}}">
+                <a id="delete-{{this.id}}" class="pop-up-button_user-page">
+                    <img id="delete-img-{{this.id}}" src="/img/deleteButton.svg" class="pop-up-button-pic_user-page delete-payment-card" alt="delete button">
```
кого удаляем-то? из alt не ясно

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.hbs:
```diff
@@ -0,0 +1,49 @@
+<form method="post" class="popup-form_add-address" novalidate>
```
тут тоже классы не по БЭМу

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.js:
```diff
@@ -0,0 +1,91 @@
+            PopUp.replaceChildren();
+        }
+        if (PopUpFade) {
+            PopUpFade.style.display = 'none';
```
может ввести класс и его накидывать и снимать? а то везде инлайн-стили фиксите, прям много где

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.js:
```diff
@@ -0,0 +1,91 @@
+     * Метод, добавляющий слушатели.
+     */
+    startEventListener() {
+        const cancel = document.getElementById('popup-form_add-address__cancel');
```
нет проверки на элемент

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.js:
```diff
@@ -0,0 +1,91 @@
+        const cancel = document.getElementById('popup-form_add-address__cancel');
+        cancel.addEventListener('click', this.listenClickCancel);
+
+        const apply = document.getElementById('popup-form_add-address__apply');
```
нет проверки на элемент

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.js:
```diff
@@ -0,0 +1,91 @@
+        cancel.removeEventListener('click', this.listenClickCancel);
+
+        const apply = document.getElementById('.popup-form_add-address__apply');
+        apply.removeEventListener('click', this.listenClickApplyBind);
```
нет проверок на элементы

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.scss:
```diff
@@ -0,0 +1,67 @@
+    color: var(--secondary-purple);
+}
+
+.popup-form_add-address__input {
```
тоже не по БЭМу

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.scss:
```diff
@@ -0,0 +1,67 @@
+.popup-form_add-address__button {
+    height: 40px;
+    width: 100px;
+    /* padding: 8px 24px 10px; */
```
?

DPeshkoff commented on public/src/components/PopUpAddAddress/PopUpAddAddress.scss:
```diff
@@ -0,0 +1,67 @@
+    /* padding: 8px 24px 10px; */
+}
+
+.popup-form_add-address__cancel {
```
для чего этот класс нужен? 

DPeshkoff commented on public/src/components/PopUpAddPaymentCard/PopUpAddPaymentCard.js:
```diff
@@ -0,0 +1,80 @@
+        cancel.removeEventListener('click', this.listenClickCancel);
+
+        const apply = document.getElementById('.popup-form_add-card__apply');
+        apply.removeEventListener('click', this.listenClickApply);
```
тоже проверок не завезли

DPeshkoff commented on public/src/components/PopUpAddPaymentCard/PopUpAddPaymentCard.scss:
```diff
@@ -0,0 +1,74 @@
+    grid-gap: 6px;
+}
+
+.popup-form_add-card__header {
```
и тут не БЭМ

DPeshkoff commented on public/src/components/PopUpAddPaymentCard/PopUpAddPaymentCard.scss:
```diff
@@ -0,0 +1,74 @@
+.popup-form_add-card__input {
+    height: 44px;
+    width: 100%;
+    padding: 10px 14px 11px;
```
некратное

DPeshkoff commented on public/src/components/PopUpAddPaymentCard/PopUpAddPaymentCard.scss:
```diff
@@ -0,0 +1,74 @@
+    /* padding: 8px 24px 10px; */
+}
+
+.popup-form_add-card__cancel {
```
опять он! я его уже видел

1) переиспользуйте код, не надо копипастить
2) в чем смысл этого класса

DPeshkoff commented on public/src/components/PopUpChooseAddressAndPaymentCard/PopUpChooseAddressAndPaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+                    }
+                });
+            }
+            chooseAddress.style.border = '1px solid #6369D1';
```
вот такое точно я бы не стал инлайнить

DPeshkoff commented on public/src/components/PopUpChooseAddressAndPaymentCard/PopUpChooseAddressAndPaymentCard.js:
```diff
@@ -0,0 +1,99 @@
+        if (fields) {
+            fields.forEach((key) => {
+                const fieldId = key.getAttribute('id');
+                key.addEventListener('click', this.listenClickAddressAndPaymentCard.bind(null, fieldId));
```
я бы не стал байндить тут - лучше сделать это в другом месте, переопределив this.listenClickAddressAndPaymentCard и только затем его вешать как обработчик

DPeshkoff commented on public/src/components/PopUpChooseAddressAndPaymentCard/PopUpChooseAddressAndPaymentCard.scss:
```diff
@@ -0,0 +1,60 @@
+.cart-popup-form__button {
+    height: 40px;
+    width: 100px;
+    /* padding: 8px 24px 10px; */
```
чую очередную копипасту
может это вынести в какой-нибудь PopUpCommon.scss?

DPeshkoff commented on public/src/components/PopUpEditUserInfo/PopUpEditUserInfo.hbs:
```diff
@@ -0,0 +1,20 @@
+                <div class="field-name_user-info text-normal-default">
+                    {{this.name}}
+                </div>
+                <input maxlength="41" class="popup-form_user-info__input text-normal-default" name="{{this.name}}" id="{{this.id}}"
```
специфично. почему 41?

tUnknownLegend replied:
В инпут форме так красивее всего было

DPeshkoff commented on public/src/components/PopUpEditUserInfo/PopUpEditUserInfo.js:
```diff
@@ -0,0 +1,124 @@
+            fields: {
+                field1: {
+                    name: context.getAttribute('name'),
+                    id: context.id + '__popUp',
```
кажется, в таких местах было бы нагляднее через обратные кавычки
но это уже вкусовщина

DPeshkoff commented on public/src/components/PopUpEditUserInfo/PopUpEditUserInfo.js:
```diff
@@ -0,0 +1,124 @@
+            },
+        };
+
+        switch (context.id) {
```
вообще прикольный конструктор получился
непонятно, почему не шаблонизатором, а js сделали
в чем была идея именно такой реализации?

DPeshkoff commented on public/src/components/PopUpEditUserInfo/PopUpEditUserInfo.js:
```diff
@@ -0,0 +1,124 @@
+
+        const apply = document.getElementById('popup-form_user-info__apply');
+        this.listenClickApplyBind = this.listenClickApply.bind(this);
+        apply.addEventListener('click', this.listenClickApplyBind);
```
ну тут сами поймете, что не так)

DPeshkoff commented on public/src/components/PopUpEditUserInfo/PopUpEditUserInfo.scss:
```diff
@@ -0,0 +1,55 @@
+    margin-left: 340px;
+    margin-top: 20px;
+    width: 128px;
+}
```
в этом файле все замечания - аналогичны

DPeshkoff commented on public/src/config.js:
```diff
@@ -0,0 +1,54 @@
+
+export const config = {
+    basePath: 'https://www.reazon.ru',
+    // basePath: 'http://127.0.0.1:8080',
```
?

DPeshkoff commented on public/src/config.js:
```diff
@@ -0,0 +1,54 @@
+        cart: '/cart',
+    },
+    api: {
+        login: basePathApi + 'login',
```
вот здесь точно бы строки с обратными кавычками запилил

DPeshkoff commented on public/src/index.js:
```diff
@@ -58,52 +23,37 @@ const changePage = async (event) => {
+    if (href === config.href.logout) {
+        event.preventDefault();
+        userActions.logout();
     }
```
почему здесь логика роутера лежит?

DPeshkoff commented on public/src/index.js:
```diff
@@ -58,52 +23,37 @@ const changePage = async (event) => {
-        config.auth.authorised = true;
-        router.remove(config.header.login.href);
-        router.remove(config.header.signup.href);
+userStore.addListener(() => {
+    if (userStore.getContext(userStore._storeNames.responseCode) === 200) {
+        refresh.onLogOut();
```
тоже не понял, что это здесь такое
можно же экшон разлогина бросать

tUnknownLegend replied:
так он слушает этот экшон и отрегивает страницы роутера

DPeshkoff replied:
а почему он это делает вне диспетчера? не лучше ли сделать сущность обновлятора, которая будет подписана на стор и действовать в той же системе координат?

DPeshkoff commented on public/src/modules/ErrorMessage.js:
```diff
@@ -31,34 +39,74 @@ class ErrorMessage {
     /**
      * Метод, который отрисовывает сообщение об ошибке сервера.
-     * @param {object} target - HTML-элемент, после которого будет осуществлена отрисовка
+     * @param {object} target - HTML-элемент, после (до) которого будет осуществлена отрисовка
```
object?

DPeshkoff commented on public/src/modules/ErrorMessage.js:
```diff
@@ -31,34 +39,74 @@ class ErrorMessage {
+                    <span class="server-error__text" id="server-error__text_">
+                        ${errorText}
+                    </span>
+                        </div>`);
```
а почему не шаблон?

DPeshkoff commented on public/src/modules/ErrorMessage.js:
```diff
@@ -31,34 +39,74 @@ class ErrorMessage {
         if (!valData.status) {
+            const exError = document.getElementById('error-text');
+            if (exError) {
+                if (!(exError.innerText === valData.message)) {
```
можно объединить условия

DPeshkoff commented on public/src/modules/Router.js:
```diff
@@ -53,62 +58,78 @@ class Router {
+        this.register(config.href.signup, RegisterPage);
+        this.register(config.href.category, CatalogPage);
+        this.register(config.href.cart, CartPage);
+        // this.register(config.href.user, UserPage); //fix
```
так чего там насчет фикса?)

tUnknownLegend replied:
это просто надо убрать, но для тестов нужно, чтобы можно было на локалке в профиль перейти

DPeshkoff commented on public/src/modules/Router.js:
```diff
@@ -53,62 +58,78 @@ class Router {
-            this.#currentPage = this.#pathToPage.get(path)(config);
+    openPage(path) {
+        let goToPath = path?.slice(0, path.lastIndexOf('/'));
+        goToPath = goToPath ? goToPath : path;
```
а в чем смысл? можно константу в одной строчке определять с проверкой...

tUnknownLegend replied:
сдеал одной строчкой, но не уверен что так лучше, ?? не получается тут использовать, так как пустая стринга возвращается :(

DPeshkoff commented on public/src/modules/ajax.js:
```diff
@@ -1,12 +1,14 @@
         'Content-Type': 'application/json',
         'accept': 'application/json',
-        'Origin': 'https://www.reazon.ru',
+        // 'Origin': 'https://www.reazon.ru',
+        // 'Origin': 'http://localhost:8081',
```
в итоге у вас нет никакого ориджина
зачем безопасность убили? к требованиям каждого рк написано "не сломать предыдущие рк"

tUnknownLegend replied:
да, надо раскомментить

DPeshkoff commented on public/src/modules/ajax.js:
```diff
@@ -50,7 +52,25 @@ class Request {
+     * @param {Blob} data - полезная нагрузка запроса
+     * @return {Promise<Response>} промис запроса
+     */
+    makePostRequestSendAva = async (url, data) => {
```
ava = )

DPeshkoff commented on public/src/modules/refreshElements.js:
```diff
@@ -33,14 +39,14 @@ export default class RefreshEl {
     /**
      * Метод, реализующий перерендеринг компонента Header
-     * @param {object} config - контекст отрисовки компонента
+     * @param {boolean} auth - контекст отрисовки компонента
```
неправда, это не контекст, это auth

DPeshkoff commented on public/src/modules/sharedFunctions.js:
```diff
@@ -0,0 +1,38 @@
+/**
+ * Класс с функциями общего назначения
+ */
+class SharedFunctions {
```
самый главный вопрос

а почему это класс? эти функции ничего между собой не объединяет

если уж хочется сделать модуль с sharedFunctions - экспортируйте все эти функции и просто импортируйте нужные

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+     * Функция, регистрирующая листенеры сторов
+     */
+    addListener() {
+        cartStore.addListener(this.getCart.bind(this), CartActionTypes.GET_CART);
```
bind лучше отдельно делать

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+    deleteItem(id) {
+        cartAction.deleteById(id);
+        const deleteElement = document.getElementById(`cart-item_cart/${id}`);
+        deleteElement.remove();
```
каждый свой день я начинаю с попытки remove'ать undefined)

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+        [context.sumPrice, context.noSalePrice, context.priceDiff, context.count] =
+            data.reduce((sumVal, key) => {
+                // sumPrice
+                sumVal[0] += key.lowprice * key.count ?? key.price * key.count;
```
ничего не понятно
но интересно

к таким сложным элементам нужно бы побольше комментариев - что конкретно здесь мы редьюсим и зачем это сделано именно так (а не через объект с 4 полями)

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+                });
+            }
+            context.isAuth = userStore.getContext(userStore._storeNames.isAuth);
+            context.isAuth = true; // FIX
```
ISWYDT

давайте пофиксим

tUnknownLegend replied:
тут все норм - опять же чисто для тестов на локалке

DPeshkoff replied:
на проде не раскомменчено, где же норм-то

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+            cartItem.render(data);
+            this.startEventListener();
+        } else {
+            document.getElementById('main').innerHTML = `
+            <div class="paint-background"></div>
+            <div id="content-cart"
+                <span class="text-normal-large-normal cart-main_empty">
+                Корзина пуста. Случайно не нужен&nbsp
+                <a href="${config.href.category}/phones" class="link">телефон</a>
+                ?
+                </span>
+            </div>
+            <div class="paint-background"></div>`;
```
Почему не шаблон?

tUnknownLegend replied:
Не хотелось уже в шаблон лезть, проще было так сделать, да и в целом, если успеем надо бы на jsx перейти...

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+     */
+    getUserData() {
+        switch (itemsStore.getContext(itemsStore._storeNames.responseCode)) {
+        case config.responseCodes.code200:
+            break;
+        case config.responseCodes.code401:
+            break;
```
а где загрузка-то? не очень понятно

tUnknownLegend replied:
Там они сразу подставляются без ответа сервера, а если сервер 500 кинул, то показываем ошибку

DPeshkoff replied:
довольно неочевидная система, если честно

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+     * Функция, обрабатывающая клики на данной странице
+     * @param {HTMLElement} event контекст события для обработки
+     */
+    async listenClickAddressAndPaymentCardBlock(event) {
```
switch внутри switch это красиво, но давайте это декомпозируем

tUnknownLegend replied:
Надо сделать базовый попап и от него наследоваться, пока этого нет проблемы и css и с этим... Но мы сделаем :)

DPeshkoff replied:
а когда?)

tUnknownLegend replied:
не знаю, пока не успеваю :\ Вынес второй свитч, чтобы не так страшно выглядело

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+                        document.getElementById(`sale-price/${itemId}`).textContent);
+                    const count = sharedFunctions._parseInt(
+                        document.getElementById(`count-product/${itemId}`).textContent);
+                    if (isNaN(price)) {
```
а почему не number.isNaN? в чем разница?

tUnknownLegend replied:
`isNaN` еще проверяет может ли переменная быть конвертирована в число. Посклоьку у нас точно должно быть число, то корректнее твой вариант

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+            });
+        }
+        // Подсчет итоговой стоимости товаров в корзине для отрисовки
+        const summary = {};
+        this.#calcSummaryPrice(data, summary);
+        // Изменение итоговых сумм
+        const totalPrice = document.getElementById('total-price');
+        totalPrice.textContent = summary.sumPrice + ' ₽';
+        const productsNumber = document.getElementById('products-number');
+        productsNumber.textContent = 'Товары, ' + summary.count + ' шт.';
+        const priceWithoutDiscount = document.getElementById('price-without-discount');
+        priceWithoutDiscount.textContent = summary.noSalePrice + ' ₽';
+        const discount = document.getElementById('discount');
+        discount.textContent = summary.priceDiff + ' ₽';
```
выглядит, как задача для шаблонизатора

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+            let time = document.getElementById('time-delivery').textContent.trim();
+            date = date.split(' / ');
+            time = time.split(' - ');
+            orderData.deliveryDate = new Date(Date.UTC(date[2], date[1], date[0],
+                (Number(time[1].split(':')[0]) + Number(time[0].split(':')[0])) / 2 % 24,
+                0)).toJSON();
```
что здесь происходит? почему так сложно, нет пояснений и не вынесено в utils какой-нибудь?

tUnknownLegend replied:
Этом мы из двух полей ввода парсим время и дату и делаем из них Date

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+     */
+    #getDate(firstDayIn) {
+        const getDate = (next) => {
+            const currDate = new Date(new Date().getTime() + next * 24 * 60 * 60 * 1000);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setDate

DPeshkoff commented on public/src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,525 @@
+    #getDate(firstDayIn) {
+        const getDate = (next) => {
+            const currDate = new Date(new Date().getTime() + next * 24 * 60 * 60 * 1000);
+            return `${currDate.getDate()} / ${currDate.getMonth()} / ${currDate.getFullYear()}`;
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString

DPeshkoff commented on public/src/pages/CartPage/CartPage.scss:
```diff
@@ -0,0 +1,374 @@
+
+.checkbox-opt {
+  width: 3ex;
+  height: 3ex;
```
почему решили взять ex?)

DPeshkoff commented on public/src/pages/CartPage/CartPage.scss:
```diff
@@ -0,0 +1,374 @@
+}
+
+
+/* Pseudo element for check styling */
+
```
поясните, плиз, что происходит))

thelvv replied:
Кастомизация чекбоксов

DPeshkoff commented on public/src/pages/CartPage/CartPage.scss:
```diff
@@ -0,0 +1,374 @@
+  cursor: pointer;
+}
+
+.option_select:hover {
```
не по БЭМу, кстати

DPeshkoff commented on public/src/pages/CartPage/CartPage.scss:
```diff
@@ -0,0 +1,374 @@
+.select-opt {
+  opacity: 0;
+  position: absolute;
+  left: -40000px;
```
ну ничего себе

tUnknownLegend replied:
то чего пользовтель не видит - того и нет ;)

DPeshkoff commented on public/src/pages/CartPage/CartPage.scss:
```diff
@@ -0,0 +1,374 @@
+  opacity: 1;
+}
+
+.select:focus .select-opt:checked + .option_select {
```
сложные селекторы - зло, лишнее время на обработку css

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.hbs:
```diff
@@ -0,0 +1,33 @@
+    <div class="catalog_category-name text-normal-huge">{{this.category}}</div>
+
+    <div class="catalog_filters-items-block">
+        <div>
+
+        </div>
```
для чего этот див?

thelvv replied:
Чтобы оставить текущую ячейку грида пустой

DPeshkoff replied:
костыльно

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.js:
```diff
@@ -0,0 +1,309 @@
+     * Функция, реагирующая на кнопки сортировки.
+     *  @param {HTMLElement} event - событие, вызвавшее клик.
+     */
+    lisitenSortCatalog(event) {
```
опечатка

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.js:
```diff
@@ -0,0 +1,309 @@
+
+    /**
+     * Функция, реагирующая на кнопки сортировки.
+     *  @param {HTMLElement} event - событие, вызвавшее клик.
```
событие это не HTMLElement

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.js:
```diff
@@ -0,0 +1,309 @@
+     * Функция, обрабатывающая скролл на странице
+     */
+    bottomOfPageHandlerPrototype() {
+        if ((scrollY + innerHeight > (0.8 * document.body.scrollHeight))) {
```
магия вне Хогвартса требует пояснений в лице комментариев

tUnknownLegend replied:
смотрим, если положение скролла меньше чем 0.8 длинны страницы

DPeshkoff replied:
я к тому, что в таких местах нужны дополнительные комментарии в коде)

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.js:
```diff
@@ -0,0 +1,309 @@
+     */
+    bottomOfPageHandlerPrototype() {
+        if ((scrollY + innerHeight > (0.8 * document.body.scrollHeight))) {
+            if (!this.waitThrottleScroll) {
```
а спиннер загрузки есть? 

tUnknownLegend replied:
не очень понял что есть спинер

tUnknownLegend replied:
Теперь понял, нет

DPeshkoff commented on public/src/pages/CatalogPage/CatalogPage.scss:
```diff
@@ -0,0 +1,64 @@
+  color: var(--jet-black);
+}
+
+.catalog_filter-block__body {
```
не БЭМ

DPeshkoff commented on public/src/pages/LoginPage/LoginPage.hbs:
```diff
@@ -12,7 +12,7 @@
     </div>
     <div class="right paint-purple">
-        <img type="image" class="login-right-img" src="./img/LoginRightImg.png" alt="LoginRightImg">
+        <img type="image" class="login-right-img" src="/img/LoginRightImg.webp" alt="LoginRightImg">
```
alt не верный

DPeshkoff commented on public/src/pages/LoginPage/LoginPage.js:
```diff
@@ -1,11 +1,13 @@
+import refresh from '../../modules/refreshElements.js';
 
 const ERROR_400_MESSAGE = 'Ошибка. Попробуйте еще раз';
 const ERROR_401_MESSAGE = 'Неверная почта или пароль';
```
в файлик с константами

DPeshkoff commented on public/src/pages/LoginPage/LoginPage.js:
```diff
@@ -50,14 +28,52 @@ export default class LoginPage extends BasePage {
+            !document.getElementById('serverErrorMessage') ?
+                errorMessage.getServerMessage(document.getElementById('inForm'),
+                    'serverErrorMessage', SERVER_ERROR_MESSAGE) :
+                console.log('server error: ', status);
```
у вас есть всплывашка ошибок, давайте сюда ее встроим

DPeshkoff commented on public/src/pages/MainPage/MainPage.js:
```diff
@@ -51,63 +25,216 @@ export default class MainPage extends BasePage {
+     */
+    localEventListenersHandler(event) {
+        event.preventDefault();
+        if (event.target.getAttribute('ind')) {
```
можно же было сделать дата-атрибут

DPeshkoff commented on public/src/pages/MainPage/MainPage.js:
```diff
@@ -51,63 +25,216 @@ export default class MainPage extends BasePage {
+                '[ind=\'itemcard_item-count\/' +
+                cartStore.getContext(cartStore._storeNames.currID) + '\']');
+            if (itemCount) {
+                itemCount.forEach((item) => item.textContent = '1');
```
выглядит очень костыльно

DPeshkoff commented on public/src/pages/RegisterPage/RegisterPage.hbs:
```diff
@@ -12,7 +12,7 @@
     </div>
     <div class="paint-purple">
-        <img type="image" class="LoginRightImg" src="./img/LoginRightImg.png" alt="LoginRightImg">
+        <img type="image" class="LoginRightImg" src="/img/LoginRightImg.webp" alt="LoginRightImg">
```
alt

DPeshkoff commented on public/src/pages/RegisterPage/RegisterPage.js:
```diff
@@ -1,11 +1,13 @@
+import refresh from '../../modules/refreshElements.js';
 
 const ERROR_400_MESSAGE = 'Ошибка. Попробуйте еще раз';
 const ERROR_401_MESSAGE = 'Неверная почта или пароль';
```
в константы, дублирование кода

DPeshkoff commented on public/src/pages/RegisterPage/RegisterPage.js:
```diff
@@ -66,14 +28,52 @@ export default class RegisterPage extends BasePage {
+        switch (status) {
+        case 201:
+            refresh.onAuth();
+            router.openPage(config.href.main); // fix change to prev
```
что за фикс?)

tUnknownLegend replied:
чтобы на прошлую страницу переходило в истории - никак руки не доходят

DPeshkoff replied:
довольно критично - давайте чинить

DPeshkoff commented on public/src/pages/UserPage/UserPage.hbs:
```diff
@@ -0,0 +1,104 @@
+
+            <div class="change-pop-up_user-page change-user-photo_user-page" id="change-user-photo_user-page">
+                <input type="file" style="display: none;" id="changeUserPhoto__upload"
+                       accept="image/jpeg, image/png, image/jpg">
```
а webp?

DPeshkoff commented on public/src/pages/UserPage/UserPage.hbs:
```diff
@@ -0,0 +1,104 @@
+                    <span id="name-text">Нет информации</span>
+                {{/if}}
+                <a class="edit-profile-data">
+                    <img class="edit-button_user-page" src="/img/editButton.svg" alt="edit button">
```
alt - кого эдитим?

DPeshkoff commented on public/src/pages/UserPage/UserPage.hbs:
```diff
@@ -0,0 +1,104 @@
+                <div class="separation-line_user-page"></div>
+                <div class="pop-up-button_user-page">
+                    <img src="/img/deleteButton.svg" id="deleteUserPhoto" class="pop-up-button_user-page-pic"
+                         alt="delete button">
```
alt - кого эдитим и делитим?

DPeshkoff commented on public/src/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,421 @@
+        userStore.addListener(this.templateFunction.bind(this, this.renderPaymentCards.bind(this)),
+            ProfileActionTypes.SAVE_ADD_CARD);
+
+        userStore.addListener(this.templateFunction.bind(this, this.renderAddresses.bind(this)),
```
bind внутри bind... не очень читабельно

DPeshkoff commented on public/src/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,421 @@
+     * Функция, для передачи в листнер стора
+     * @param {function} toDo - обработчик события
+     */
+    templateFunction(toDo) {
```
а проверить, точно ли это функция?

DPeshkoff commented on public/src/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,421 @@
+                paymentCard.addEventListener('mouseleave', this.listenMouseOutPaymentCard);
+            });
+        } else {
+            console.log('element not found', this.paymentCard);
```
лишнее

DPeshkoff commented on public/src/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,421 @@
+                addressCard.addEventListener('mouseleave', this.listenMouseOutAddressCard);
+            });
+        } else {
+            console.log('element not found', this.addressCard);
```
лишнее

DPeshkoff commented on public/src/pages/UserPage/UserPage.js:
```diff
@@ -0,0 +1,421 @@
+    /**
+     * Метод, добавляющий слушатели.
+     */
+    startEventListener() {
```
console log лишние

DPeshkoff commented on public/src/pages/UserPage/UserPage.scss:
```diff
@@ -0,0 +1,170 @@
+  @extend .payment-cards-block_user-page;
+}
+
+.payment-cards-items_user-page {
```
не БЭМ

DPeshkoff commented on public/src/pages/UserPage/UserPage.scss:
```diff
@@ -0,0 +1,170 @@
+.card-block-title_user-page {
+  margin-bottom: -6px;
+  margin-left: 36px;
+  //align-self: end;
```
остался мусор

DPeshkoff commented on public/src/pages/UserPage/UserPage.scss:
```diff
@@ -0,0 +1,170 @@
+
+.change-user-photo_user-page {
+  width: 84px;
+  margin-top: -90px;
```
чем обусловлена такая радость? 

DPeshkoff commented on public/src/pages/UserPage/UserPage.scss:
```diff
@@ -0,0 +1,170 @@
+}
+
+.pop-up-button_user-page {
+  //align-self: center;
```
что-то лишнее

DPeshkoff commented on public/src/pages/UserPage/UserPage.scss:
```diff
@@ -0,0 +1,170 @@
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
```
https://developer.mozilla.org/en-US/docs/Web/CSS/inset

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+            count: 2,
+            name: `Apple iPhone 13 64 ГБ \\r
+            gladwehaveanunderstanding, fuck out the way
+yeah, all your shit lame, I feel no pain, we" "\\eof`,
+            imgsrc: './img/Smartphone.webp',
+            category: '',
+            price: 100000,
+            lowprice: 80000,
+            id: 12,
+            rating: 10,
+        },
+    ];
+
+    #data = {
+        addressID: 1111,
+        city: 'Москва',
+        street: 'Мира',
+        house: 15,
+        flat: 4,
+        deliveryPrice: 'Бесплатно',
+        date: new Date('2022-11-25'),
+        // paymentMethodProvider: mirIcon,
+        avatar: './img/Smartphone.webp',
+        username: 'Джахар',
+        phone: '+7 (872) 234-23-65',
+        deliveryDate: this.#getDate(1),
+        deliveryTime: '18:00 - 23:00',
+        cardNumber: '8765432143212546',
+        expiry: '05 / 24',
+        paymentCardId: 1,
+        auth: true,
+    };
+
+    _storeNames = {
+        responseCode: 'responseCode',
+        itemsCart: 'itemsCart',
+        cartID: 'cartID',
+        userID: 'userID',
+        currID: 'currID',
+    };
```
на фронте такого быть не должно

tUnknownLegend replied:
это моки для отладки, они не используются на проде

DPeshkoff replied:
ну и давайте выпилим тогда

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+
+        const itemsCart = this._storage.get(this._storeNames.itemsCart);
+        console.log('itemsCart', itemsCart);
+        console.log('response', response);
```
дебаг-логи

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+        });
+        console.log('merge', itemsCart);
+
+        if (status === config.responseCodes.code200) {
```
а если не 200?

tUnknownLegend replied:
если не 200 - ничего менять не будем, просто сообщение об ошибке (надеюсь) покажем

DPeshkoff replied:
давайте прямо ситуацию обработаем)

tUnknownLegend replied:
ошибка же во view отображается, поэтому написал надеюсь, а так 100% что обрабоатем)

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+                items: itemsCart.map(({id}) => id),
+            }).catch((err) => console.log(err));
+            this._storage.set(this._storeNames.responseCode, postStatus);
+            if (postStatus === config.responseCodes.code200) {
```
а если не 200?

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+            .catch((err) => console.log(err));
+
+        this._storage.set(this._storeNames.responseCode, status);
+        if (status === config.responseCodes.code200) {
```
а если не 200?

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+     */
+    async _getCart() {
+        const [status, response] = await request.makeGetRequest(config.api.cart)
+            .catch((err) => console.log(err));
```
так-то у вас оно здесь должно обработаться, но зачем так странно статусы обрабатывать?

tUnknownLegend replied:
Ты про `.catch((err) => console.log(err))`? Или про `status`? статусы все кладутся в мапку `this._storage.set(this._storeNames.responseCode, status);` и потом во вьюшке чекаются. `catch` тут на всякий, так как промис возвращяется.

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+            if (item.id === id) {
+                delete itemsCart[key];
+            }
+        });
```
filter? find?

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+     * Действие: добавить товар в корзину.
+     * @param {int} status
+     * @param {int} countChange
+     * @param {int} id
```
что за инты

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,330 @@
+            return `${currDate.getDate()} / ${currDate.getMonth()} / ${currDate.getFullYear()}`;
+        };
+        return Array.from(Array(7).keys()).map((inDays) => getDate(inDays + firstDayIn));
+    }
```
ну вот так и знал, что дублирование будет)
выносите в utils

DPeshkoff commented on public/src/stores/ItemsStore.js:
```diff
@@ -0,0 +1,248 @@
+ * Класс, реализующий базовое хранилище.
+ */
+class ItemsStore extends BaseStore {
+    #topCategory = {
+        Smartphone: {
+            nameCategory: 'Телефоны',
+            img: './img/Smartphone.webp',
+            href: config.href.category + '/phones',
+        },
+        Computer: {
+            nameCategory: 'Компьютеры',
+            img: './img/Computer.webp',
+            href: config.href.category + '/computers',
+        },
+        Monitors: {
+            nameCategory: 'Мониторы',
+            img: './img/Monitors.webp',
+            href: '/category/monitors',
+        },
+        TV: {
+            nameCategory: 'Телевизоры',
+            img: './img/TV.webp',
+            href: '/category/televisors',
+        },
+        Watch: {
+            nameCategory: 'Часы',
+            img: './img/Watch.webp',
+            href: '/category/watches',
+        },
+        Tablet: {
+            nameCategory: 'Планшеты',
+            img: './img/Tablet.webp',
+            href: config.href.category + '/tablets',
+        },
+        Accessories: {
+            nameCategory: 'Аксессуары',
+            img: './img/Accessories.webp',
+            href: '/category/accessories',
+        },
+    };
```
этих товарищей тут быть тоже не должно, по-хорошему...

DPeshkoff commented on public/src/stores/ItemsStore.js:
```diff
@@ -0,0 +1,248 @@
+   * @param {boolean} isFirstRequest - получали ли мы до этого карточки
+   */
+    _getPopularItemCard(isFirstRequest) {
+    }
```
пустое(

DPeshkoff commented on public/src/stores/ItemsStore.js:
```diff
@@ -0,0 +1,248 @@
+    #syncWithCart(items) {
+        const cartItems = cartStore.getContext(cartStore._storeNames.itemsCart);
+        if (cartItems) {
+            items.forEach((item) => {
+                cartItems.forEach((cartItem) => {
```
o(n^2) или сколько тут?

tUnknownLegend replied:
получается что так, но пока в голову не приходит как переписать :\

DPeshkoff commented on public/src/stores/ItemsStore.js:
```diff
@@ -0,0 +1,248 @@
+          )}&count=${5}&${window.location.search.substring(1)}`,
+            )
+            .catch((err) => console.log(err));
+        this._storage.set(this._storeNames.responseCode, status);
```
нечитабельно совсем
вынесите это куда-нибудь, пожалуйста, в переменные или методы

DPeshkoff commented on server/index.js:
```diff
@@ -11,11 +11,11 @@ app.use(express.static(publicFolder));
-});
+// app.get('/sw.js', (req, res) => {
+//     res.sendFile(path.resolve(`${publicFolder}/sw.js`));
+// });
```
а зачем убили?

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+import {cartAction} from '../actions/cart';
+
+/**
+ * Класс, реализующий базовое хранилище.
```
опечатка в названии файла

tUnknownLegend replied:
что-то не вижу опечатки, вроде корзина, все верно

DPeshkoff replied:
userst_r_ore

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+        },
+    };
+
+    #testPaymentCards = [
+        {
+            id: 60,
+            type: '',
+            number: '1232131232131231',
+            expiryDate: '2053-01-31T00:00:00Z',
+            priority: false,
+        },
+        {
+            number: '123456******5678',
+            type: 'MIR',
+            expiry: '1987-08-19T23:15:30.000Z',
+            id: 2,
+        },
+    ];
+
+    #testAddressCards = [
+        {
+            priority: true,
+            city: 'г. Москва',
+            street: 'улица Бауманская',
+            house: '228',
+            flat: '420',
+            id: 1, // addressCard/
+        },
+        {
+            city: 'г. Москва',
+            street: 'улица Бассейная',
+            house: '228',
+            flat: '420',
+            id: 2,
+        },
+    ];
```
что за тестовые данные?

tUnknownLegend replied:
опять же для отладки

DPeshkoff replied:
выпиливаем)

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+                    }, 'userpage__popUp__error');
+                break;
+            case this.#context.fields.phone.popUpName:
+                isValid &= errorMessage.validateField(validation.validatePhone(value),
```
чем обусловлен такой оператор? выглядит так, будто у вас что-то не то с архитектурой

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+     * @param {object} data - данные для обработки
+     * @return {string} errorMessage - сообщение об ошибке
+     */
+    #validateCard(data) {
```
кажется, валидация обитает в modules/validation

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+     * @param {object} data - данные для обработки
+     * @return {string} errorMessage - сообщение об ошибке
+     */
+    #validateAddress(data) {
```
аналогично

DPeshkoff commented on public/src/stores/UserStrore.js:
```diff
@@ -0,0 +1,626 @@
+
+    /**
+     * Метод, реализующий удаление карты адреса.
+     * @param {int} id - идентификатор элемента
```
int

DPeshkoff reviewed (changes requested):
Посмотрел код

tUnknownLegend committed:
сделал базовый компонент для карточки userpage

tUnknownLegend committed:
убрал моки

tUnknownLegend committed:
stylelint

tUnknownLegend committed:
Merge branch 'Base-2.17-merge_fix' into Base-2.14_merge_cart

tUnknownLegend committed:
исправления пустой корзины

tUnknownLegend committed:
исправления пустой корзины

tUnknownLegend committed:
исправил удаление 1 элемента

tUnknownLegend committed:
исправил соедение локально корзины с серверной

tUnknownLegend committed:
исрпавление загрузки карт

tUnknownLegend committed:
исправление загрузк товаров

tUnknownLegend committed:
stylint config fix

tUnknownLegend committed:
Merge branch 'Base-2.17-merge_fix' into Base-2.14_merge_cart

tUnknownLegend committed:
Update webpack.config.js

thelvv committed:
фикс стилей в соответствии с линтером

thelvv committed:
частичная правка помарок из ПР

thelvv committed:
промежуточный фикс ПР

thelvv committed:
стили приведены к БЭМ, поправлены альты

thelvv committed:
удаление багов, появившихся после фиксов

tUnknownLegend committed:
сделал базовый класс для попапов

нужно нормально сделать компоненты попапоа, с шаблонами, стилями и т.п. чтобы копипасты избежать. Пока на это времени нет. Так же внес другие порпавки, основанные на замечаниях в ревью

tUnknownLegend committed:
upstream merge

tUnknownLegend committed:
перенем логику роутера и индекса в роутер

так же поправил взаимодействие с history

tUnknownLegend committed:
убрал класс sharedFuncitons, сделал экспорт функций

tUnknownLegend committed:
изменил getDate

и другие исправления

tUnknownLegend committed:
переместил валидацию из стора

tUnknownLegend committed:
переделал слушатили попапа в корзине

DPeshkoff commented on public/index.scss:
```diff
@@ -26,12 +28,20 @@
+    margin: 0;
+}
+
+input[type="number"] {
```
приколы какие-то(

DPeshkoff commented on public/src/components/InfoCard/AddressCard/AddressCard.scss:
```diff
@@ -0,0 +1,94 @@
+    filter: none;
+}
+
+//.address-card:hover {
```
кто такое

DPeshkoff commented on public/src/components/popUp/PopUpAddAddress/PopUpAddAddress.scss:
```diff
@@ -0,0 +1,63 @@
+    height: 40px;
+    width: 100px;
+
+    /* padding: 8px 24px 10px; */
```
кто

DPeshkoff commented on public/src/components/popUp/PopUpAddPaymentCard/PopUpAddPaymentCard.scss:
```diff
@@ -0,0 +1,70 @@
+    height: 40px;
+    width: 100px;
+
+    /* padding: 8px 24px 10px; */
```
кто

DPeshkoff commented on public/src/components/popUp/PopUpEditUserInfo/PopUpEditUserInfo.js:
```diff
@@ -0,0 +1,91 @@
+import './PopUpEditUserInfo.scss';
+import {profileAction} from '../../../actions/profile';
+import BasePopUp from '../BasePopUp';
+import {config} from '../../../config';
```
почините ворнинг

DPeshkoff commented on public/src/components/popUp/PopUpEditUserInfo/PopUpEditUserInfo.scss:
```diff
@@ -0,0 +1,52 @@
+    height: 40px;
+    width: 100px;
+
+    /* padding: 8px 24px 10px; */
```
кто

DPeshkoff commented on public/src/modules/Router.js:
```diff
@@ -1,7 +1,14 @@
+import ErrorPage from '../pages/ErrorPage/ErrorPage';
+import {config} from '../config';
+import CartPage from '../pages/CartPage/CartPage';
+import UserPage from '../pages/UserPage/UserPage';
```
почините ворнинг

DPeshkoff commented on public/src/stores/CartStore.js:
```diff
@@ -0,0 +1,258 @@
+import {config} from '../config';
+import userStore from './UserStrore';
+import itemsStore from './ItemsStore';
+import {getDate} from '../modules/sharedFunctions';
```
поправьте ворнинг (такой лучше бы в ошибки выносить сразу)

DPeshkoff commented on webpack.config.js:
```diff
@@ -92,12 +99,20 @@ const config = {
+        new StylelintPlugin({
+            configFile: 'stylelint.config.js',
+            extensions: ['scss'],
+            // files: './public/*',
```
комментики

tUnknownLegend committed:
поправил замечания

tUnknownLegend committed:
изменил настройки линтера

tUnknownLegend committed:
вернул CSP

tUnknownLegend committed:
ci

DPeshkoff reviewed (approved):
Мне ок!
